src/xrpld/app/tx/detail/PayChan.cpp

```diff
@@ -517,7 +541,12 @@ PayChanClaim::doApply()
             return tecNO_TARGET;
 
         // Check whether the destination account requires deposit authorization.
-        if (depositAuth && (sled->getFlags() & lsfDepositAuth))
+        if (ctx_.tx.isFieldPresent(sfCredentialIDs))
+        {
+            if (credentials::removeExpired(view(), ctx_.tx, j_))
+                return tecEXPIRED;
+        }
```

Comment:
Same here (and in all the other transactors with `CredentialIDs`)

Reply:
same as above